sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ErrorParserSuite.scala

```diff
@@ -63,4 +63,46 @@ class ErrorParserSuite extends SparkFunSuite {
       "Combination of ORDER BY/SORT BY/DISTRIBUTE BY/CLUSTER BY is not supported",
       "^^^")
   }
+
+  test("hyphen in identifier - DDL tests") {
+    val msg = "unquoted identifier"
+    // ddl tests
+    intercept("use test-test", 1, 8, 9, msg + " test-test", "^^^")
```

Comment:
Since this test case is for `DDL tests`. we had better have `CREATE/DROP DATABASE` and `CREATE/DROP FUNCTION` test coverages, too. Otherwise, we will miss like https://github.com/apache/spark/pull/24749/files#r290055158 .